google/ggrc-core
# GGRC-4515 Fix 400 after assessment status and cav update

# Issue description

Updating assessment's status and cad cause response with 400 status.

# Steps to test the changes

Merge this PR with https://github.com/google/ggrc-core/pull/7551.

Have an assessment with mandatory text cad filled with value and status 'In Review'.
1) Create assessment template with mandatory local CAD with `Text` type;
2) Create an assessment; put it in `In Review` status;
3) Click 'Needs Rework' button;
4) Remove local cad value and press 'Enter';
5) Enter a new cad value and press 'Complete'.

As cad value has been changed, status should be "In Progress", cad value should be changed.

# Solution description

Validation on complete status for put has been moved to the before commit hook as it is have to be performed after object has been precessed but before commit.

# Sanity checklist




# PR Review checklist


# TODO

- [x] Investigate if validation on complete is still actual;
- [x] Investigate why status is not saved if checks passed.
